### ISSUE TITLE
Remove dependency to fastify-warning ^0.2.0

### ISSUE THREAD

**Issue by viheikki** [pinojs/pino#1269]
The fastify-warning package pino v. 7 depends on has the version number 0.2.0, which according to semver indicates it is in development and not meant to be used by external packages.
The way fastify-warning is used by pino in deprecations.js could be easily replicated with a few lines of custom code, thus removing a dependency to a pre v. 1.0.0 package.
Furthermore, the dependency in Pino is set to version ^0.2.0, meaning there is significant risk that an update to fastify-warning will break Pino, since no backwards compatibility is expected from pre v. 1.0.0 packages. Pre v. 1.0.0 packages should be **always** pegged to the exact version, if for some reason such package needs to be used.

**Comment by mcollina** [comment]
Why?

**Comment by jsumners** [comment]
I agree that `fastify-warning` should be bumped to v1.0.0. I do not agree we need to duplicate its code in order to get there.

**Comment by viheikki** [comment]
> Why?

The fastify-warning package pino v. 7 depends on has the version number 0.2.0, which according to semver indicates it is in development and not meant to be used by external packages. That should be reason enough combined with the dependency pegged to ^0.2.0 that is just begging for trouble.

**Comment by mcollina** [comment]
It's just a version number. We maintain both things so it's not going to be a problem.

**Comment by viheikki** [comment]
> It's just a version number. We maintain both things so it's not going to be a problem.

Regardless of who maintains what package, if the package is published in npm, it should follow the semver rules. (Notable exception being @types/packages and similar).
See https://semver.org/
